conventions/cli.md

```diff
@@ -64,6 +64,13 @@ In this case `foo` is positional, and refers to the service to create.
   example, `--image=img.repo/asdf` in Knative Serving sets
   `spec.template.containers[0].image`
 
+* Flags that control a boolean behavior (eg. generate a name or not) are
+  specified by their presence. The default happens when the flag is not present,
+  and adding the flag marks the user's desire for the non-default thing. When
+  the flag *disables* a default behavior which is to do something, it should
+  start with `no` (eg. `--no-generate-name` when the default is to generate a
+  name).
+
```

Comment:
Actually, both variations (with and without `--no`) should be present in this case, regardless of the default value  (using the variant that corresponds to the default value should be possible, too). In this example, e.g. `--generate-name` and `--no-generate-name` should be both available all the time (corresponding `--generate-name=true` and `--generate-name=false`).

The reason why I prefer this variant is:

* There is no ambiguity what should be used as boolean value ("true", "yes" and "on" are commonly used as such values)
* It's shorter
* It stands in the tradition of GNU getopts
* Many boolean options are modelled like this in Unix (they are called 'flags' in this case)
* We already use that style e.g. for `kn --help` (or would we like to switch to `kn --help=true` ?). Other examples: `kn service create --force` or `kn service create --async` would have change to `kn service create --force=true` and `kn service create --async=true` (or maybe in the latter case `--sync=false`)
* For consistencies reason we would always have to require options with a value.

Reply:
> Actually, both variations (with and without --no) should be present in this case, regardless of the default value (using the variant that corresponds to the default value should be possible, too).

I have the same feeling.

Reply:
just noticed your statement here:

>We already use that style e.g. for kn --help (or would we like to switch to kn --help=true ?). Other examples: kn service create --force or kn service create --async would have change to kn service create --force=true and kn service create --async=true (or maybe in the latter case --sync=false)

`--help` continues to work as cobra defaults it to mean `true`. so full form `--flag=false` is only necessary when turning things off.

Reply:
@cppforlife Have you tried `kn --help=false` ?